owners/src/file_cache.js

```diff
@@ -16,17 +16,41 @@
 
 const {CloudStorage} = require('./cloud_storage');
 
+/**
+ * A generic cache interface.
+ */
+class FileCache {
+  /**
+   * Fetch the contents of a file.
+   *
+   * @param {string} filename file to get contents of.
+   * @param {string} getContents function to get contents if file not in cache.
+   * @return {string} file contents.
+   */
+  async readFile(filename, getContents) {
+    return await getContents();
```

Comment:
Where is `getContents()` defined in this context?

Reply:
If I'm understanding your question correctly, `getContents` is provided by whatever client code is trying to read a file through the cache. The idea being, `getContents` is the "real" way to read the file (ie. filesystem IO or API call), but it may not be required if the file is in the cache.